extensions/arc/src/test/ui/dashboards/postgresOverview.test.ts

```diff
@@ -0,0 +1,113 @@
+import { FakeControllerModel } from '../../mocks/fakeControllerModel';
+import { FakeAzdataApi } from '../../mocks/fakeAzdataApi';
+import { PostgresModel } from '../../../models/postgresModel';
+import { ControllerModel, Registration } from '../../../models/controllerModel';
+
+describe('postgresOverviewPage', () => {
+	let postgresOverview: PostgresOverviewPage;
+	let azdataApi: azdataExt.IAzdataApi;
+	let controllerModel: ControllerModel;
+	let postgresModel: PostgresModel;
+
+	let showInformationMessage: sinon.SinonSpy;
+	let showErrorMessage: sinon.SinonSpy;
+
+	let informationMessageShown: Deferred;
+	let errorMessageShown: Deferred;
+
+	beforeEach(async () => {
+		// Stub the azdata CLI API
+		azdataApi = new FakeAzdataApi();
+		const azdataExt = TypeMoq.Mock.ofType<azdataExt.IExtension>();
+		azdataExt.setup(x => x.azdata).returns(() => azdataApi);
+		sinon.stub(vscode.extensions, 'getExtension').returns(<any>{ exports: azdataExt.object });
+
+		// Stub the window UI
+		informationMessageShown = new Deferred();
+		showInformationMessage = sinon.stub(vscode.window, 'showInformationMessage').callsFake(
+			(_: string, __: vscode.MessageOptions, ...___: vscode.MessageItem[]) => {
+				informationMessageShown.resolve();
+				return Promise.resolve(undefined);
+			});
+
+		errorMessageShown = new Deferred();
+		showErrorMessage = sinon.stub(vscode.window, 'showErrorMessage').callsFake(
+			(_: string, __: vscode.MessageOptions, ...___: vscode.MessageItem[]) => {
+				errorMessageShown.resolve();
+				return Promise.resolve(undefined);
+			});
+
+		// Setup the PostgresModel
+		controllerModel = new FakeControllerModel();
+		const postgresResource: PGResourceInfo = { name: 'my-pg', resourceType: '' }
+		const registration: Registration = { instanceName: '', state: '', instanceType: ResourceType.postgresInstances }
+		const treeDataProvider = new AzureArcTreeDataProvider(TypeMoq.Mock.ofType<vscode.ExtensionContext>().object);
+		postgresModel = new PostgresModel(controllerModel, postgresResource, registration, treeDataProvider);
+
+		// Setup the PostgresOverviewPage
+		const { modelViewMock } = createModelViewMock();
+		postgresOverview = new PostgresOverviewPage(modelViewMock.object, controllerModel, postgresModel);
+		postgresOverview['toolbarContainer'];
+	});
+
+	afterEach(() => {
+		sinon.restore();
+	});
+
+	describe('delete button', () => {
+		let refreshTreeNode: sinon.SinonStub;
+
+		beforeEach(() => {
+			sinon.stub(utils, 'promptForInstanceDeletion').returns(Promise.resolve(true));
+			sinon.stub(controllerModel, 'acquireAzdataSession').returns(Promise.resolve(vscode.Disposable.from()));
+			refreshTreeNode = sinon.stub(controllerModel, 'refreshTreeNode');
+		});
+
+		it('deletes Postgres on success', async () => {
+			// Stub 'azdata arc postgres server delete' to return success
+			const postgresDelete = sinon.stub();
+			sinon.stub(azdataApi, 'arc').get(() => {
```

Comment:
@Charles-Gagnon  I wanted to do:

`sinon.stub(azdataApi.arc.postgres.server, 'delete')`

to stub just the delete function.  It compiles, but didn't override the implementation.  I had to stub the entire arc() getter.

Reply:
Alright, issue should be fixed. https://github.com/microsoft/azuredatastudio/pull/14502/commits/b2391e52d27ba991496784697244f982536cac2e

The problem was that the fake API was returning a new object each time the arc getting was called - so stubbing arc.<whatever> would stub that object but not the one that the test used.

Fixed it by having there just be one object and returning that instead - which can then be stubbed as needed.

Fixed up the other tests that had this issue as well.

Reply:
Ah nice, thanks for the fix